backend/server/resources/resources.go

```diff
@@ -0,0 +1,113 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	. "csesoc.unsw.edu.au/m/v2/server"
+
+	"github.com/labstack/echo/v4"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var resourceColl *mongo.Collection
+
+// Resource - struct for the list of resources that are displayed
+type Resource struct {
+	Title       string `json:"title" validate:"required"`
+	Description string `json:"description" validate:"required"`
+	Link        string `json:"link" validate:"required,url"`
+	Source      string `json:"src" validate:"required"`
+}
+
+///////////
+// SETUP
+///////////
+
+// ResourcesSetup - Set up the resources collection
+func ResourcesSetup(client *mongo.Client) {
+	resourceColl = client.Database("csesoc").Collection("resources")
+
+	// Creating unique index for resource title
+	opt := options.Index()
+	opt.SetUnique(true)
+	index := mongo.IndexModel{
+		Keys:    bson.M{"title": 1},
+		Options: opt,
+	}
+	if _, err := resourceColl.Indexes().CreateOne(context.Background(), index); err != nil {
+		log.Fatal("Could not create index: ", err)
+	}
+
+	// Fetching resource list
+	resources, err := retrieveJSON()
+	if err != nil {
+		log.Fatal("Could not retrive resources from JSON")
+	}
+
+	for _, resource := range resources {
+		if _, err := resourceColl.InsertOne(context.TODO(), resource); err != nil {
+			log.Printf("Could not insert resource " + resource.Title + " " + err.Error())
+		}
+	}
+}
+
+///////////
+// HANDLERS
+///////////
+
+// GetPreview godoc
+// @Summary Get a list of resources stored
+// @Tags resources
+// @Param tier query integer false "Valid sponsor tier, 0-2 inclusive" mininum(0) maxinum(2)
```

Comment:
Change documentation references to `sponsor`.

Reply:
Whoops missed that. I also did the swag init and it added events as well so guess we missed that too.